js/validation.js

```diff
@@ -0,0 +1,65 @@
+import {formElement, HOUSING_MIN_PRICE} from './data.js';
+const adTitleInput = formElement.querySelector('#title');
+const typeSelect = formElement.querySelector('#type');
+const priceInput = formElement.querySelector('#price');
+const countRoomsSelect = formElement.querySelector('#room_number');
+const countGuestsSelect = formElement.querySelector('#capacity');
+
+const valuePriceCheck = () => {
+  priceInput.min = HOUSING_MIN_PRICE[typeSelect.value];
+  (Number(priceInput.value) < Number(priceInput.min)) ? priceInput.setCustomValidity(`Минимальная цена за ночь ${  priceInput.min}`) : priceInput.setCustomValidity('');
+  priceInput.reportValidity();
+};
+
+adTitleInput.addEventListener('input', () => {
+  const valueLength = adTitleInput.value.length;
+  const valueMin = adTitleInput.attributes.minlength.value;
+  const valueMax = adTitleInput.attributes.maxlength.value;
+  if (valueLength < valueMin) {
+    adTitleInput.setCustomValidity(`Ещё ${valueMin - valueLength} симв.`);
+  }
+  else if (valueLength > valueMax) {
+    adTitleInput.setCustomValidity(`Удалите лишние ${valueLength - valueMax} симв.`);
+  }
+  else {
+    adTitleInput.setCustomValidity('');
+  }
+  adTitleInput.reportValidity();
+});
+
+typeSelect.addEventListener('change', () => {
+  priceInput.placeholder = String(HOUSING_MIN_PRICE[typeSelect.value]);
+  if(priceInput.value.length > 0) {
+    valuePriceCheck();
+  }
+});
+
+priceInput.addEventListener('input', () => {
+  valuePriceCheck();
+});
+
+countRoomsSelect.addEventListener('change', (e) => {
```

Comment:
Элемент `form` содержит в себе ссылки на все инпуты. Если инпуты именованы, то хорошим решением было бы вынести форму в переменную и обращаться к ее инпутам через их `name`:
```suggestion
adForm['room_number'].addEventListener('change', (e) => {
// если поле содежит в себе только буквы, то можно обращаться через точечную нотацию: adForm.timein
```
Так мы улучшим читаемость и производительность кода и не будем создавать лишних переменных